dist/PreserveModuleNamePlugin.js

```diff
@@ -73,6 +73,29 @@ function getPreservedModules(modules) {
                 m[exports.preserveModuleName] = req;
                 return true;
             }
+            // Since its relative, link it with the module alias if possible
+            const reasonRequest = r.module && r.module.rawRequest;
+            const moduleDependency = r.dependency && r.dependency.module;
+            const moduleRequest = moduleDependency && removeLoaders(moduleDependency.request);
+            if (req && aliases && reasonRequest && moduleRequest) {
+                const aliasMatches = Object.keys(aliases).filter((alias) => {
+                    const aliasPath = aliases[alias];
+                    const matchPath = !!(moduleRequest.match(new RegExp(`^${aliasPath}`)));
```

Comment:
Is the regexp totally safe? Did you consider if `aliasPath` might need escaping? For example, what if it contains dots or dollars?
Be careful that Webpack also has a convention where an alias ending with `$` must be an exact match (which is what it would mean in a regexp, so that's ok at least).